sql/fishing_contest_entries.sql

```diff
@@ -0,0 +1,21 @@
+--
+-- Table structure for table `fishing_contest_entries`
+-- This table holds the submissions to the fish ranking contest
+--
+
+CREATE TABLE IF NOT EXISTS `fishing_contest_entries` (
+    `name`        varchar(15)             NOT NULL,
+    `mjob`        tinyint(2)  unsigned    NOT NULL,
+    `sjob`        tinyint(2)  unsigned    NOT NULL,
+    `mlevel`      tinyint(2)  unsigned    NOT NULL,
+    `slevel`      tinyint(2)  unsigned    NOT NULL,
+    `race`        tinyint(2)  unsigned    NOT NULL,
+    `allegiance`  tinyint(2)  unsigned    NOT NULL,
+    `fishrank`    tinyint(2)  unsigned    NOT NULL,
+    `score`       int(4)      unsigned    NOT NULL DEFAULT 0,
+    `submittime`  int(4)      unsigned    NOT NULL DEFAULT 0,
+    `contestrank` tinyint(2)  unsigned    NOT NULL DEFAULT 0,
+    `share`       tinyint(2)  unsigned    NOT NULL DEFAULT 0,
+    `claimed`     tinyint(1)  unsigned    NOT NULL DEFAULT 0,
+    PRIMARY KEY (`name`) -- Only one entry per character
+) ENGINE=MyISAM DEFAULT CHARSET=utf8;
```

Comment:
Is there a reason why a lot of this data isn't pulled from the submitting char with a JOIN?

Reply:
Also 'name' isn't necessarily unique. IDK if that could lead to any issues, but probably want to use charid instead and a join like Zach said.

Reply:
Only done this way for consistency with the transmitted packet.  Can very easily be a join.  However, if a character gets deleted, their contest entry should remain intact on the leaderboard, by name.

Reply:
The issue I'm seeing is that if a character is deleted it'll still exist in the chars table, and a new char could be created with the same name. Would that prevent the new character from also entering the contest?

Reply:
> The issue I'm seeing is that if a character is deleted it'll still exist in the chars table, and a new char could be created with the same name. Would that prevent the new character from also entering the contest?

In theory yes, but that's a case I'm not sure how retail handles.  I could simply remove the entry if the character is deleted.

Reply:
As I just played around with this, I think it's far more straightforward to just use the name in the table rather than the charid.  If I use the ID, I need to do another table lookup any time it gets references as the player is not guaranteed to be online if another player pulls up the list, for example.  I'd be more worried about lookup errors using the ID.

Reply:
Not sure about retail either, and it's definitely a fringe issue. Plus I was made aware that we don't actually even open a name back up for reuse on deletion yet, but should in the future. I would use charid as primary key since that for sure will never be changed or reused under normal circumstances and just pull all of the char data that is currently here with a join instead, including name.

Reply:
> If I use the ID, I need to do another table lookup any time it gets references as the player is not guaranteed to be online if another player pulls up the list, for example. I'd be more worried about lookup errors using the ID.

Not sure what you mean. Player being online or not doesn't change anything about char lookup by charid.